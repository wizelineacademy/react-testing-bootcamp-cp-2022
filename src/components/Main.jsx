import React, { useEffect, useState } from "react";
import { POD } from "./POD";
const API_KEY = "aLF9OyI9eUKScjU0GIazR3BKJqntNfay4NXmE6lT";

const style = {
  background: "#A0BCC2",
  height: "60vh",
  maxWidth: "89vw",
  margin: "1vh auto",
  padding: "1vh",
};

const inputStyle = {
  display: "block",
  margin: "1rem auto",
};

export const Main = () => {
  const [formatedDate, setFormatedDate] = useState("");
  const [pic, setPic] = useState(null);
  const [date, setDate] = useState(new Date().toISOString().substring(0, 10));
  const [url, setUrl] = useState(
    `https://api.nasa.gov/planetary/apod?api_key=${API_KEY}`
  );

  useEffect(() => {
    if (date !== new Date().toISOString().substring(0, 10)) {
      setUrl(
        `https://api.nasa.gov/planetary/apod?api_key=${API_KEY}&date=${date}`
      );
    } else {
      setUrl(`https://api.nasa.gov/planetary/apod?api_key=${API_KEY}`);
    }
  }, [date]);

  useEffect(() => {
      setPic(null);
    fetch(url)
      .then((data) => data.json())
      .then((data) => {
        setPic(data);
        setFormatedDate(
          new Date(data.date).toLocaleDateString("es-MX", {
            day: "2-digit",
            month: "2-digit",
            year: "numeric",
          })
        );
      })
      .catch((err) => console.log(err));
  }, [url]);

  return (
    <div style={style}>
      <input
        type="date"
        name="date"
        id="date"
        data-testid="dateInput"
        style={inputStyle}
        value={date}
        onChange={(event) => {
          setDate(new Date(event.target.value).toISOString().substring(0, 10));
        }}
      />
      { pic === null && 
        <POD />
      }
      {pic !== null && (
        <POD
          title={pic.title}
          date={formatedDate}
          url={pic.url}
          explanation={pic.explanation}
        />
      )}
    </div>
  );
};
