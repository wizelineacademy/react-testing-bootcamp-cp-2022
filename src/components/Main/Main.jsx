import React, { useState, useEffect, useCallback } from "react";
import Style from "./Main.module.css";
import { Row, Col, DatePicker, Image, Spin, notification } from "antd";
import moment from "moment";

const Main = () => {
  const [loading, setLoading] = useState(false);
  const [date, setDate] = useState(null);
  const [image, setImage] = useState(null);
  const [title, setTitle] = useState("");
  const [infoImage, setInfoImage] = useState(null);
  const { REACT_APP_API_KEY_NASA_POD, REACT_APP_API_URL } = process.env;

  const onChangeDate = useCallback(
    async (datemoment, datestring) => {
      setLoading(true);
      setDate(datemoment);
      try {
        const response = await fetch(
          `${REACT_APP_API_URL}?api_key=${REACT_APP_API_KEY_NASA_POD}&date=${datestring}`
        );
        const POD = await response.json();
        updateStates(POD);
        if (POD.msg) {
          sendNotificationError(POD);
        }
      } catch (error) {
        sendNotificationError("There was an error, please try again.");
      }
      setLoading(false);
    },
    [REACT_APP_API_KEY_NASA_POD, REACT_APP_API_URL]
  );

  const sendNotificationError = ({ msg = "" }) => {
    notification["error"]({
      message: "Error",
      description: msg,
    });
  };

  const updateStates = ({ url = "", explanation = "", title = "" }) => {
    setImage(url);
    setInfoImage(explanation);
    setTitle(title);
  };

  useEffect(() => {
    const today = moment();
    onChangeDate(today, today.format("YYYY-MM-DD"));
  }, [onChangeDate]);

  return (
    <div className={Style.container}>
      <Row>
        <Col span={6} offset={9} className={Style.calendar}>
          <DatePicker onChange={onChangeDate} value={date} />
        </Col>
      </Row>
      <Spin
        tip="Loading..."
        spinning={loading}
        wrapperClassName={Style.spinner}
      >
        <Row>
          <Col span={10} offset={1}>
            {image && <Image src={image} alt={title} />}
          </Col>
          <Col span={8} offset={2}>
            {infoImage && <p>{infoImage}</p>}
          </Col>
        </Row>
      </Spin>
    </div>
  );
};

export default Main;
