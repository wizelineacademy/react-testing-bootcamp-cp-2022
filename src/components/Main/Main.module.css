.container {
  width: 100%;
  flex: auto;
  background-color: #ffffff;
  display: flex;
  flex-direction: column;
}

.calendar {
  width: 100%;
  display: flex;
  justify-content: center;
  padding-bottom: 25px;
  padding-top: 25px;
}

.infoContainer {
  display: flex;
  background-color: blue;
}

.image,
.text {
  width: 50%;
}

.spinner {
  height: 100%;
}