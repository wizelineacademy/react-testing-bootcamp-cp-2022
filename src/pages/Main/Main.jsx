import Footer from "../../components/Footer";
import Header from "../../components/Header";
import Content from "../../components/Content";

const Main = () => (
	<>
		<Header />
		<Content />
		<Footer />
	</>
);

export default Main;
