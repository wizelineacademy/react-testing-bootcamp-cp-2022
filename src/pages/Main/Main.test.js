import { render, screen } from "@testing-library/react";

import Main from ".";

describe("Main page", () => {
	it("should render header and footer", () => {
		render(<Main />);

		const header = screen.getByRole("heading", {
			name: /nasa's picture of the day/i,
		});
		const footer = screen.getByText(
			/project created during wizeline academy react testing bootcamp/i
		);

		expect(header).toBeInTheDocument();
		expect(footer).toBeInTheDocument();
	});
});
