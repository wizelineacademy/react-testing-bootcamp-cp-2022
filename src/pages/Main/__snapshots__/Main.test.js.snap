// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page should render header, content, and footer 1`] = `
<div>
  <header>
    <h1
      class="sc-gsnTZi bKySKi"
    >
      Nasa's picture of the day
    </h1>
  </header>
  <div
    class="sc-dkzDqf cvGcvx"
  >
    <input
      class="sc-hKMtZM gsOFdg"
      data-testid="date-field"
      type="date"
      value="2022-04-28"
    />
    <button
      class="sc-eCYdqJ dNVnSQ"
    >
      Show
    </button>
  </div>
  <p>
    Loading...
  </p>
  <footer>
    <p
      class="sc-bczRLJ hAGsLM"
    >
      Project created during Wizeline Academy React Testing Bootcamp
    </p>
  </footer>
</div>
`;
